File: go/app/health/health.go
package health

import (
	"github.com/gin-gonic/gin"
	"net/http"
)

type healthCheck struct {
	Status string `json:"status"`
}

func Started(c *gin.Context) {
	c.JSON(http.StatusOK, &healthCheck{Status: "OK"})
}

func Live(c *gin.Context) {
	c.JSON(http.StatusOK, &healthCheck{Status: "OK"})
}

func Ready(c *gin.Context) {
	c.JSON(http.StatusOK, &healthCheck{Status: "OK"})
}

File: rust/src/health.rs
use rocket::serde::{json::Json};
use serde::{Serialize};

#[derive(Serialize)]
pub struct HealthCheck {
    status: String,
}

#[rocket::get("/started")]
pub fn started() -> Json<HealthCheck> {
    Json(HealthCheck { status: String::from("OK") })
}

#[rocket::get("/live")]
pub fn live() -> Json<HealthCheck> {
    Json(HealthCheck { status: String::from("OK") })
}

#[rocket::get("/ready")]
pub fn ready() -> Json<HealthCheck> {
    Json(HealthCheck { status: String::from("OK") })
}
